### ISSUE TITLE
flags: 'g' not working

### ISSUE THREAD

**Issue by inzanez** [Va1/string-replace-loader#24]
Hi

I've been trying to get string-replace-loader to work with the 'g' flag.

...

>         loader: 'string-replace-loader',
>         query: {
>           search: 'REST_API',
>           replace: 'test',
>           flags: 'g'
>         }

That doesn't seem to work out. If I remove the 'g' flag and set '' or 'i' as flag, it works perfectly. But 'g' flag always brings back the following error:
```
ERROR in ./~/css-loader?sourceMap!./~/vue-loader/lib/style-compiler?{"id":"data-v-84da6350","scoped":false,"hasInlineConfig":false}!./~/vue-loader/lib/selector.js?type=styles&index=0!./~/string-replace-loader?{"search":{},"replace":"test","flags":"g"}!./src/components/something/Test.vue
Module build failed: TypeError: Cannot read property 'content' of undefined
    at Object.module.exports (/home/inzanez/WebDevel/testapp/node_modules/vue-loader/lib/selector.js:18:27)
 @ ./~/vue-style-loader!./~/css-loader?sourceMap!./~/vue-loader/lib/style-compiler?{"id":"data-v-84da6350","scoped":false,"hasInlineConfig":false}!./~/vue-loader/lib/selector.js?type=styles&index=0!./~/string-replace-loader?{"search":{},"replace":"test","flags":"g"}!./src/components/something/Test.vue 4:14-409 13:3-17:5 14:22-417
 @ ./src/components/something/Test.vue
 @ ./src/main.js
 @ multi (webpack)-dev-server/client?http://localhost:8080 webpack/hot/dev-server ./src/main.js
```
Any idea what might be wrong here?oO

**Comment by derekshull** [comment]
+1 on this.

```
    {
        test: /\.(js|vue)$/,
        loader: 'string-replace-loader',
        enforce: 'pre',
        query: {
          search: '__API__',
          replace: 'https://localhost:8082',
          flags: 'i'
        }
      }
```

**Comment by inzanez** [comment]
@derekshull: Maybe as a workaround:
node_modules/string-replace-loader/index.js:
I changed line 7:
`options.search = new RegExp(options.search, options.flags);`
to
`options.search = new RegExp('__API__', 'g');`

I know it's not a fix but it works for meantime ;-)

**Comment by shirishp** [comment]
I have created a new issue with a possible workaround - https://github.com/Va1/string-replace-loader/issues/31

**Comment by cdll** [comment]
+1 on this.
i've also tried
````
        query: {
          search: '@@basepath'
          ,replace: './public'
          ,flags: 'g'
        }
````
and 
````
        query: {
          multiple:[
            {
              search: '@@basepath'
              ,replace: './'
              ,flags: 'g'
            }
        }
````
both failed

**Comment by beautifulBoys** [comment]
everyone:  try this "webpack-replace-loader" . 
this loader wrote by me.

**Comment by Va1** [comment]
sorry for the delayed reply. this was caused by the bug when options being overwritten, while they should remain immutable. fixed in v2.0.0 released recently
